data_handling/controllers/articles.controllers.js

```diff
@@ -35,14 +35,21 @@ exports.getArticles = (req, res, next) => {
 };
 
 exports.getArticlesById = (req, res, next) => {
-    const {id} = req.params
+    const {comments} = req.query;
+    const {id} = req.params;
 
     returnCommentCount().then((countData) => {
```

Comment:
Instead of returnCommentCount being a separate function you can do all of this in one query. I can't see this function on this commit but it can all be done within fetchArticlesById. When you use SELECT * it automatically selects all of the colums, you can name which ones you want and in turn, create new columns. So you can do something like SELECT article.id, article.title, COUNT(comments) FROM etc...
The key thing to add would be a JOIN so you have access to the comments table and the articles table!